textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexMatcher.java

```diff
@@ -190,9 +189,38 @@ private void javaRegexMatch(String fieldValue, String fieldName, List<Span> span
 	private void re2jRegexMatch(String fieldValue, String fieldName, List<Span> spanList) {
 		com.google.re2j.Matcher re2jMatcher = this.re2jPattern.matcher(fieldValue);
 		while (re2jMatcher.find()) {
-			spanList.add(new Span(fieldName, re2jMatcher.start(), re2jMatcher.end(), this.regexPredicate.getRegex(), fieldValue));
+			int start = re2jMatcher.start();
+			int end = re2jMatcher.end();
+			spanList.add(new Span(fieldName, start, end, 
+					this.regexPredicate.getRegex(), fieldValue.substring(start, end)));
+		}
+	}
+	
+	public void setUseJavaRegex() throws java.util.regex.PatternSyntaxException {
```

Comment:
Yep. Currently regex matcher is set to RE2J by default. But considering Java Regex is always faster, we can change to Java Regex by default. But I think it's good to leave a setter function anyway.